### DISCUSSION TITLE
Questions on choosing target variable

### DISCUSSION THREAD

**Discussion by LoZnzz** [winedarksea/AutoTS#194] [Q&A]
Hi,

First I would say I am pretty new to timeseries prediction and I would like to thank you all for buliding up this helpful package, I really love the idea of automating the process so as to save effort.

So when I was looking into the docs, I got some questions regarding on how AutoTS chooses the target varibales in a df with more than one feature. I tried to use create_regressor to make regressor but that doesn't seem appropriate as it is neither letting me to dedicate a target variable nor asking me to provide the range of features used to faciliate the prediction of target variable. 

I would like to ask if I should use some other methods to dedicate a target variable or I simply just feed AutoTS with the dataframe and let it decide which one is? 

Thanks.

**Comment by winedarksea** [comment]
This is a common question, because things are different here than a basic machine learning regression and it gets confusing. You should not be providing hand-crafted features here like you might for an ML model, only inputting information, some of which will be `covariates` and some of which will be `future_regressors`.

Your input `df` is a selection of time series. Some of these can be `covariates` which are additional series providing information. So your target might be 'house prices' and a covariate series could be 'interest rates'. To make it clear to the model which is your priority, there is a `weights` arg to `.fit` which can accept a dictionary weighting the one you are most concerned about forecasting. Only _multivariate_ models can make use of covariates.

Models, both multivariate and univariate, can also use `future_regressors`. I use the term "future" to make it clear that these exogenous features should be known about in advance. So future_regressor passed to .predict are already known. This type of data, known for the future, is pretty rare. One way around this is using `create_regressor` which lags the historical features forwards, making it usable as a future regressor (albeit not as great as having actually known the future values for it) and also tacks on some holiday features and such.

Your best bet is to start with just passing your time series of interest in to the dataframe, then adding in your extra covariates next. Follow the production_example.py and examine the shape and index of the input data

**Reply by LoZnzz** [reply]
Thanks for your kind reply!
